src/omnicore/fetchwallettx.cpp

```diff
@@ -0,0 +1,111 @@
+// The fetch functions provide a sorted list of transaction hashes ordered by block, position in block and position in wallet including STO receipts
```

Comment:
Hehe - there are similarities for sure but the context and methods are different... 

Basically this is a "getter" function for wallet history (note there is no cache) - it just abstracts all the stuff around ```OrderedTxItems``` from the each separate function usage into a central function that also handles the extra stuff for pending and STO...

From now on anything that needs access to history can simply call ```FetchWalletOmniTransactions()``` and get an ordered list of Omni transactions to work with - it was mainly about a) simplifying and b) removing code duplication - eg if we now look at ```listtransactions_MP```, it used to be messy function now it's as simple as:
```
    // obtain a sorted list of Omni layer wallet transactions (including STO receipts and pending)
    std::map<std::string,uint256> walletTransactions = FetchWalletOmniTransactions(nFrom+nCount, nStartBlock, nEndBlock);

    // reverse iterate over (now ordered) transactions and populate RPC objects for each one
    Array response;
    for (std::map<std::string,uint256>::reverse_iterator it = walletTransactions.rbegin(); it != walletTransactions.rend(); it++) {
        uint256 txHash = it->second;
        Object txobj;
        int populateResult = populateRPCTransactionObject(txHash, txobj, addressParam);
        if (0 == populateResult) response.push_back(txobj);
    }
```

Same goes for the UI, I was able to remove a bunch of crap from the tx history dialog, and am about to do the same for the trade history dialog.  As we scale into more features and other RPC calls and UI elements start to require access to wallet history I felt moving to a single/central function to provide the data was useful :)

Hope that explains a bit :)